### DISCUSSION TITLE
Discussion of data mapping (LinkML Transformer) requirements and approach

### DISCUSSION THREAD

**Discussion by cmungall** [linkml/linkml#1829] [Ideas]
This discussion is for anything relating to the LinkML transformer, including the need for a transformer, and whether it should be prioritized.

Anyone is welcome to comment, but it's recommended you read some existing background material:

- the README here https://github.com/linkml/linkml-transformer
- The Tutorial here https://github.com/linkml/linkml-transformer/blob/main/notebooks/Tutorial.ipynb
-  Mungall, C. (2023, April 23). Mapping Data Structures: Challenges and Approaches. International Society for Biocuration, Padua, Italy. Zenodo. https://doi.org/10.5281/zenodo.10343505

**Comment by dalito** [comment]
It looks like transformers could also handle migrations when schemas evolve. Was that also a motivation?

**Reply by cmungall** [reply]
Great point!

Yes, this was one of the original motivations, but not really fully articulated, see slide 7 from this (very rough) early draft of some ideas:
https://docs.google.com/presentation/d/1rP3QEU4H0f-s4TLFufAHtQPl-v0u_UgZx6yJvxm9bCk/edit#slide=id.g214d529df46_0_0

**Comment by diatomsRcool** [comment]
I think LinkML transformer is 100% needed.
